data/lua/explode_ability.lua

```diff
@@ -0,0 +1,56 @@
+
+local _ = wesnoth.textdomain 'wesnoth-help'
+local T = wml.tag
+
+-- The explode event code
+wesnoth.game_events.add_repeating("die", function()
```

Comment:
I think this should use `game_events.add` instead of `add_repeating`. Then there will be no need for those early returns, because the function will only be called if the filters pass.

Reply:
I don't understand the advantage, can you explain?

Reply:
`add_repeating` adds an event with no filters – your code will be called every time any unit dies, even if it's not an ant. With `add` you can also specify filters – the code will only be called when an ant dies (or other unit with the ability).

Functionally, I don't _think_ there's much difference, since it _is_ a repeating event. I think it's stylistically better, though.